src/models/abstract_model.py

```diff
@@ -3,5 +3,46 @@
 class AbstractModel(metaclass=abc.ABCMeta):
     @classmethod
     def __subclasshook__(cls, subclass):
```

Comment:
Hay algún motivo para usar subclasshook aquí?

Reply:
For a good overview of why, check the accepted answer to: https://stackoverflow.com/questions/40764347/python-subclasscheck-subclasshook